WIP: Add JSON-RPC interface documentation

promag opened the pull request:
This is poorly written and structured but I want to know if it makes sense.

The idea is not to copy `-help` outputs.

Closes #14448.

Follow ups:
 - explain named parameters
 - detail how to select the wallet, how -rpcwallet works, etc*
 - explain how deprecation works
 - https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-456202424 
 - https://github.com/bitcoin/bitcoin/pull/14458#discussion_r257926917
 - add related configurations while taking into account https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-454432287

kristapsk commented:
Concept ACK

ch4ot1c commented:
Concept ACK 👍

sipa commented:
Concept ACK

jnewbery commented on doc/JSON-RPC-interface.md:
```diff
@@ -0,0 +1,105 @@
+
+The JSON-RPC API is by default enabled in bitcoind and disabled on bitcoin-qt.
+This can be changed with `-noserver` and `-server` options. In bitcoin-qt it is
+possible to issue execute RPC methods in the Debug Console Dialog.
```
Remove 'issue '

jnewbery reviewed (commented):
Concept ACK. Looks good so far!

practicalswift commented on doc/JSON-RPC-interface.md:
```diff
@@ -0,0 +1,105 @@
+JSON-RPC Interface
```
Please rename this file `json-rpc-interface.md` to make it consistent with the other files.

ch4ot1c replied:
I think matching `REST-interface.md` is the intention - should it be lowercase too?

practicalswift commented on doc/JSON-RPC-interface.md:
```diff
@@ -0,0 +1,105 @@
+
+*To see the available methods use `bitcoin-cli -regtest help`*
+
+#### Server Configuration
```
Increment header levels one level at a time. You might want to use `mdl` to check the Markdown formatting :-)

laanwj commented:
@harding might want to have a look here as this adds technical documentation

promag commented:
Rebased, still WIP though.

MarcoFalke commented on doc/JSON-RPC-interface.md:
```diff
@@ -2,8 +2,8 @@
 `bitcoin-qt` has it disabled by default. This can be changed with the `-server`
-option. In the GUI it is possible to execute RPC methods in the Debug Console
-Dialog.
+or `-noserver` options. In the GUI it is possible to execute RPC methods in the
```
noserver is not a separate option

fanquake commented on doc/JSON-RPC-interface.md:
```diff
@@ -36,3 +36,113 @@ However, the wallet may not be up-to-date with the current state of the mempool
+
+## Basic usage
+
+Launch the daemin in the background:
```
s/daemin/daemon

Sjors replied:
Nit / later: explain that as an alternative you can run the daemon in one tab and use bitcoin-cli in another. That way you can keep an eye on logs. Or just mention `tail -f <datadir>/<testnet3 or regtest>/debug.log`

See also #15438.

Sjors commented on doc/JSON-RPC-interface.md:
````diff
@@ -36,3 +36,113 @@ However, the wallet may not be up-to-date with the current state of the mempool
+     corresponding -wallet option passed to bitcoind)
+```
+
+*Also talk about named parameters?*
````
This should go away if we merge it for for the 0.18 release.

Sjors commented:
ACK. I suggest we remove the "todo" comment, merge this and improve it after the 0.18 release.

Sjors reviewed (commented):
ACK. I suggest we remove the "todo" comments, merge this and improve it after the 0.18 release.

promag committed:
doc: Improve JSON-RCP interaface documentation

promag commented:
Rebased and removed configurations section, kind of agree with @Sjors https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-454432287. What left isn't worth I think, so closing.
